### DISCUSSION TITLE
How to create DataTable with 2D array in PrimeVUE

### DISCUSSION THREAD

**Discussion by Weicai-Ling** [primefaces/community#287] [PrimeVue Templates]
Hello, 

Who could please advise how to create a DataTable with 2D array fetched from server? Must the 2D array to be converted into Primevue datatable type to create a DataTable ?

Thanks,
Forrest

**Comment by Weicai-Ling** [comment]
Who could please help to advise an example code to create a DataTable with 2D array in primeVUE ?

**Comment by tugcekucukoglu** [comment]
We cannot know the exact data that you want to use. Therefore, you will have complete control.

**Reply by Weicai-Ling** [reply]
Thanks. I completed this by assign a variable for the Colunm and dataTable with the data fetched from server and put it in onMounted method. and create the dataTable with dynamic columns approach.
 
For example: 

onMounted(() => {
    productService.getProductsSmall().then((data) => {
        columns.value = data[0];
        tableData.value = data.slice(1).map(row => columns.value.reduce((acc, col, i) => {
            acc[col] = row[i];
            return acc;
        }, {}));
    });
});

Thanks for your reply.
